### PR TITLE
DOCTEAM-1338:Setting up a systemd service-article

### PR DESCRIPTION
**Pull request by Amrita42**
### Description

The scope of this PR is to cover the scope of "Setting up a systemd service" as defined in https://jira.suse.com/browse/DOCTEAM-1338

**Reviewers**
Please dont review now , will tag in slack when ready.
 
**Reused** (Review not required as this topics have been reviewed and are being re-used multiple times)
- What is systemd?
- Structure of a unit file
- Unit file types
- Unit dependencies and order

**Review required**
- Creating a Linux service with systemd
- Editing a systemd unit file
- Debugging a systemd service

### CODE REVIEW

**Comment by Amrita42** [comment]
[setting-up-systemd_en.pdf](https://github.com/user-attachments/files/17979017/setting-up-systemd_en.pdf)

File: references/systemd-editing-unit-file.xml [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+  <screen>&prompt.sudo; systemctl edit testhttp.service</screen>
+  <para>This creates a blank file that is used to override or add directives to the unit file definition.
+A directory is created in <filename>/etc/systemd/system</filename> which contains the name of the unit file
+with <literal>.d</literal> appended. For example, a directory called <literal>testhttp.service.d</literal> is created.
+  </para>
+<para>Within the directory, a snippet called <literal>override.conf</literal> is created. &systemd; merges
+    the override snippet with the full unit file, when the unit is loaded. This snippet's directives take precedence
+    over those directives in the original unit file.
+  </para>
+  <para>You can edit the full unit file instead of creating a snippet with the <literal>--full</literal>flag. For example:
```
Comment (taroth21) [inline references/systemd-editing-unit-file.xml:66]: This sentence is easy to misread (happened to me!) because 'with the --full flag' stands very close to 'creating a snippet' (so it could be understood as if one creates a snippet with the --full flag, which is of course contradictory but it took me a second read to disentangle this wrong meaning). If you move the info about the --full flag to the beginning of the sentence, there is no doubt that it refers to 'you can edit the full unit file'.

Comment (taroth21) [inline references/systemd-editing-unit-file.xml:66]:
```suggestion
  <para>With the <literal>--full</literal>flag, you can edit the full unit file instead of creating a snippet . For example:
```

File: references/systemd-debug-failed-service.xml [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+    </abstract>
+  </info>
+<para>When &systemd; fails to start a service, a generic error message is displayed, for example:</para>
+<screen>&prompt.root;<command>systemctl</command> start apache2
+Job for apache2.service failed because the control process exited with error code.
+See "systemctl status apache2.service" and "journalctl -xe" for details.</screen>
+        <variablelist>
+          <varlistentry>
+                      <term>
+List all available services and their current status:
```
Comment (taroth21) [inline references/systemd-debug-failed-service.xml:34]: Maybe this is just a matter of taste again, but I would leave out the colons a the end of the `<term>s` as you have turned the sentences into a kind of headline now by moving them into a `<term>` element

File: references/systemd-debug-failed-service.xml [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+\u25cf amavis.service                    not-found inactive dead    amavis.serv>
+  apparmor.service                       loaded    active   exited  Load AppArm>
+  appstream-sync-cache.service           loaded    inactive dead    Synchronize>
+  auditd.service                         loaded    active   running Security Au>
+  augenrules.service                     loaded    active   exited  auditd rule>
+</screen>
+      </listitem>
+          </varlistentry>
+          <varlistentry>
+             <term> If you think a specific service has failed, you can use the <literal>is-failed</literal>option:</term>
```
Comment (taroth21) [inline references/systemd-debug-failed-service.xml:56]: Maybe shorten this to: 'Check whether a certain service has failed' (as it is kind of a headline now)

File: references/systemd-debug-failed-service.xml [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+ Process: 2491 ExecStart=/usr/bin/apache2 --add-runtime oci=/usr/sbin/apache-runc $DOCKER_NETWORK_OPTIONS $DOCKER_OPTS >
+Main PID: 2491 (code=exited, status=1/FAILURE)
+</screen>
+                 <para> In the above example, the service ran as process ID 2491, but failed. Error messages give you a hint on what to do.
+                 </para>
+                 </listitem>
+          </varlistentry>
+          <varlistentry>
+          <term>
+          You can also use the <command>journalctl </command> command to check the log of the failed service:</term>
```
Comment (taroth21) [inline references/systemd-debug-failed-service.xml:80]: Same here, maybe shorten the `<term>` to: 'Check the logs of the failed service'

File: references/systemd-debug-failed-service.xml (hunk 3 above)
Comment (taroth21) [reply references/systemd-debug-failed-service.xml:56]: If you want to keep the info 'to find the specific error message', you can still provide it within an additional sentence in a `<para>` within the `<listitem>`

**Review by taroth21 — DISMISSED** [review]
@Amrita42 : Many thanks, looks good, just some minor nitpicks. Please pass on the PR to @dariavladykina for review when done!

File: references/systemd-editing-unit-file.xml (hunk 1 above)
Comment (Amrita42) [reply references/systemd-editing-unit-file.xml:66]: Yes , you are right +1 , I have corrected that

**Comment by Amrita42** [comment]
@dariavladykina  ptal thanks

File: tasks/systemd-create-service.xml [hunk 5]
```diff
@@ -35,7 +38,7 @@
       specific to the system. When you place the script in this location,you make is accessible to all system users
     without a need to specify the full path. </para>
```
Comment (dariavladykina) [inline tasks/systemd-create-service.xml:39]:
```suggestion
      specific to the system. By placing the script in this location, 
      you make is accessible to all system users
    without a need to specify the full path. </para>
```

File: tasks/systemd-create-service.xml [hunk 6]
```diff
@@ -55,20 +58,21 @@ ExecStart=/usr/local/bin/<replaceable>FILE_NAME.sh</replaceable>
...
               The <literal>Install</literal> section specifies when the service should start.
               </para>
                 </step>
                 <step><para>Save and exit the file.</para></step>
                 <step><para>To make &systemd; aware of the new service, run:
                   </para>
-                <screen>&prompt.sudo; systemctl daemon-reload</screen>
+                <screen>&prompt.sudo; systemctl <replaceable>SERVICE_NAME</replaceable></screen>
                 </step>
                 <step><para>Start,enable, and check the status of the service:</para>
```
Comment (dariavladykina) [inline tasks/systemd-create-service.xml:71]:
```suggestion
                <step><para>Start, enable and check the status of the service:</para>
```

File: references/systemd-editing-unit-file.xml [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+<!-- metadata is dealt with on the assembly level -->
+<topic xml:id="systemctl-edit-service-file"
+ role="reference" xml:lang="en"
+ xmlns="http://docbook.org/ns/docbook" version="5.2"
+ xmlns:its="http://www.w3.org/2005/11/its"
+ xmlns:xi="http://www.w3.org/2001/XInclude"
+ xmlns:xlink="http://www.w3.org/1999/xlink"
+ xmlns:trans="http://docbook.org/ns/transclusion">
+  <info>
+    <title>The <command>sysctemctl edit </command> command </title>
```
Comment (dariavladykina) [inline references/systemd-editing-unit-file.xml:19]:
```suggestion
    <title>The <command>systemctl edit</command> command </title>
```

File: references/systemd-editing-unit-file.xml [hunk 8]
```diff
@@ -0,0 +1,77 @@
...
+      <term><filename>/etc/systemd/system/</filename></term>
+      <listitem>
+        <para>
+         &systemd; unit files that are created by the <command>systemctl enable</command> command and also unit files added for extending a service.
+         This directory takes precedence over the directory with unit files created at run time.
+        </para>
+      </listitem>
+    </varlistentry>
+  </variablelist>
+  <para>The <command>sysctemctl edit </command> command by default opens a unit file snippet, for example:</para>
```
Comment (dariavladykina) [inline references/systemd-editing-unit-file.xml:56]:
```suggestion
  <para>By default, the <command>systemctl edit</command> 
  command opens a unit file snippet, for example:</para>
```

File: references/systemd-editing-unit-file.xml [hunk 9]
```diff
@@ -0,0 +1,77 @@
...
+  <screen>&prompt.sudo; systemctl edit testhttp.service</screen>
+  <para>This creates a blank file that is used to override or add directives to the unit file definition.
+A directory is created in <filename>/etc/systemd/system</filename> which contains the name of the unit file
```
Comment (dariavladykina) [inline references/systemd-editing-unit-file.xml:59]:
```suggestion
A directory is created in <filename>/etc/systemd/system</filename>, which contains the name of the unit file
```

File: references/systemd-editing-unit-file.xml [hunk 10]
```diff
@@ -0,0 +1,77 @@
...
+with <literal>.d</literal> appended. For example, a directory called <literal>testhttp.service.d</literal> is created.
+  </para>
+<para>Within the directory, a snippet called <literal>override.conf</literal> is created. &systemd; merges
+    the override snippet with the full unit file, when the unit is loaded. This snippet's directives take precedence
```
Comment (dariavladykina) [inline references/systemd-editing-unit-file.xml:63]:
```suggestion
    the override snippet with the full unit file when the unit is loaded. This snippet's directives take precedence
```

File: references/systemd-editing-unit-file.xml [hunk 11]
```diff
@@ -0,0 +1,77 @@
...
+    over those directives in the original unit file.
+  </para>
+  <para>With the <literal>--full</literal>flag, you can edit the full unit file instead of creating a snippet. For example:
```
Comment (dariavladykina) [inline references/systemd-editing-unit-file.xml:66]:
```suggestion
  <para>With the <literal>--full</literal> flag, you can edit the full unit file instead of creating a snippet. For example:
```

File: references/systemd-editing-unit-file.xml [hunk 12]
```diff
@@ -0,0 +1,77 @@
...
+    </para>
+    <screen>&prompt.sudo; systemctl edit --full testhttp.service</screen>
+<para>This loads the current unit file into the editor, where you can modify. When you save and exit the editor,the modified
+  file is written to <filename>/etc/systemd/system</filename>, which takes precedence over the system's unit definition usually located
+  in <filename>/lib/systemd/system</filename>.
```
Comment (dariavladykina) [inline references/systemd-editing-unit-file.xml:71]:
```suggestion
<para>This loads the current unit file into the editor, allowing you
to make modifications. When you save and exit the editor, the 
modified file is written to <filename>/etc/systemd/system</filename>, 
which takes precedence over the system's unit definition, usually located
in <filename>/lib/systemd/system</filename>.
```

File: references/systemd-editing-unit-file.xml [hunk 13]
```diff
@@ -0,0 +1,77 @@
...
+</para>
+<para>To remove any additions you have made, you can either delete the unit's <literal>.d</literal> configuration directory or the
+  the modified service file from <filename>/etc/systemd/system</filename>. After deleting the file or directory, reload the &systemd; process
+  so that it reverts back to the initial system process.
```
Comment (dariavladykina) [inline references/systemd-editing-unit-file.xml:75]:
```suggestion
<para>To remove any additions you have made, you can 
delete the unit's <literal>.d</literal> configuration directory or the
modified service file from <filename>/etc/systemd/system</filename>. 
After deleting the file or directory, reload the &systemd; process
so that it reverts to the initial system process.
```

**Review by dariavladykina — COMMENTED** [review]
Hi, please see some suggestions from me. Thank you!
